### PR TITLE
/utils: add `isFunction`, `findValueWith`, `makeIndexByKey`

### LINKED ISSUE #165: /utils: add `isFunction`, `findValueWith`, `makeIndexByKey`

### PR DESCRIPTION
**Pull request by mindrones**
closes #165

### CODE REVIEW

File: packages/tools/utils/src/any-boolean.js [hunk 1]
```diff
@@ -188,6 +188,39 @@ true
  */
 export const isNotNull = _.not(_.isNull);
 
+/**
+ * Return true is the input is a function
```
Comment (saabi) [inline packages/tools/utils/src/any-boolean.js]: Typo:
```
* Return true if the input is a function
              ^
```

File: packages/tools/utils/src/any-boolean.js [hunk 2]
```diff
@@ -188,6 +188,39 @@ true
...
+> function returnArgs () {return arguments}
+> isFunction(returnArgs())
+false
+ *
+ * @version 0.12.0
+ */
+export const isFunction = _.isType('Function');
+
 /**
  * Return true is the input is a number
```
Comment (saabi) [inline packages/tools/utils/src/any-boolean.js]: One here as well

**Review by saabi — CHANGES_REQUESTED** [review]
Other than the comments, looks good!

File: packages/tools/utils/src/any-boolean.js (hunk 1 above)
Comment (mindrones) [reply packages/tools/utils/src/any-boolean.js]: thanks it was spread all over the file, fixed!

**Review by saabi — APPROVED** [review]
Good to go!

**Pull request merged**